src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,421 @@
+    }
+
+    /**
+     * Assign genotypes (GTs) to the samples in the Variant Context greedily based on the PLs
+     *
+     * @param newLikelihoods       the PL array
+     * @param allelesToKeep         the list of alleles to choose from (corresponding to the PLs)
+     * @param ploidy               ploidy
+     */
+    private static void assignGenotypes(final GenotypeBuilder gb, final double[] newLikelihoods,
+                                        final List<Allele> allelesToKeep, final int ploidy) {
+        final int numNewAltAlleles = allelesToKeep.size() - 1;
+
+        // find the genotype with maximum likelihoods
+        final int indexOfLikeliestGenotype = MathUtils.maxElementIndex(newLikelihoods);
+        final GenotypeLikelihoodCalculator calculator = new GenotypeLikelihoodCalculators().getInstance(ploidy, allelesToKeep.size());
+        final GenotypeAlleleCounts alleleCounts = calculator.genotypeAlleleCountsAt(indexOfLikeliestGenotype);
+
+        gb.alleles(alleleCounts.asAlleleList(allelesToKeep));
+
+        // remove PLs if necessary in case of giant ploidy due to pooling
+        if (newLikelihoods.length > MAX_LENGTH_FOR_PL_LOGGING) {
+            gb.noPL();
+        }
+
+        if ( numNewAltAlleles > 0 ) {
+            gb.log10PError(GenotypeLikelihoods.getGQLog10FromLikelihoods(indexOfLikeliestGenotype, newLikelihoods));
+        }
+    }
+
+    /**
+     * Given set of alleles with corresponding vector of likelihoods, subset likelihoods to new set of alleles
+     *
+     * @param oldLikelihoods        Vector of PL's corresponding to original alleles
+     * @param ploidy                Ploidy (number of chromosomes describing PL's)
+     * @param originalAlleles       List of original alleles
+     * @param newAlleles            New alleles -- must be a subset of {@code originalAlleles}
+     * @return                      Vector of new PL's, ordered accorrding to GenotpyeIterator's ordering
+     */
+    private static double[] subsettedGenotypeLikelihoods(final double[] oldLikelihoods, final int ploidy,
+                                                         final List<Allele> originalAlleles, final List<Allele> newAlleles) {
+        //Since this is a private method, we will *assume* that newAlleles is a subset of originalAlleles
+        // Then we have shortcuts based on two trivial cases:
+        if (newAlleles.size() == originalAlleles.size()) {
+            return oldLikelihoods;  //no subsetting
+        } else if (newAlleles.size() == 1) {
+            return new double[] { oldLikelihoods[0]};   //hom ref is the only genotype
+        }
+
+        final int newPLSize = GLVectorSizeCache.getNumLikelihoodElements(newAlleles.size(), ploidy);
+        final double[] newPLs = new double[newPLSize];
+
+        // fill boolean array stating whether each original allele is present in new mapping
```

Comment:
I think that You may find useful the AlleleListPermutation class for some of the stuff you are doing here.